src/Trustline.ts

```diff
@@ -216,6 +220,127 @@ export class Trustline {
     })
   }
 
+  /**
+   * Prepares an ethereum transaction object for closing a trustline.
+   * @param networkAddress Address of a currency network.
+   * @param counterpartyAddress Address of counterparty to who the trustline should be settled.
+   * @param options Payment options. See `PaymentOptions` for more information.
+   * @param options.decimals Decimals of currency network can be provided manually.
+   * @param options.maximumHops Max. number of hops for transfer.
+   * @param options.maximumFees Max. transfer fees user if willing to pay.
+   */
+  public async prepareClose (
+    networkAddress: string,
+    counterpartyAddress: string,
+    options: PaymentOptions = {}
+  ): Promise<CloseTxObject> {
+    const { _user, _currencyNetwork, _transaction, _utils } = this
+
+    // Get the users options and make sure to have a decimal.
+    const { gasPrice, gasLimit, networkDecimals } = options
+    const decimals = await _currencyNetwork.getDecimals(networkAddress, { networkDecimals })
+
+    // Get close path
+    const { path, maxFees, estimatedGas, value } = await this.getClosePath(
+      networkAddress,
+      _user.address,
+      counterpartyAddress,
+      {
+        ...options,
+        networkDecimals: decimals.networkDecimals
+      }
+    )
+
+    // Make sure a path has been found.
+    if (path.length > 0) {
+      // Prepare the interaction with the contract.
+      const { rawTx, ethFees } = await _transaction.prepFuncTx(
+        _user.address,
+        networkAddress,
+        'CurrencyNetwork',
+        'closeTrustlineByTriangularTransfer',
+        [
+          counterpartyAddress,
+          _utils.convertToHexString(new BigNumber(maxFees.raw)),
+          path.slice(1)
+        ],
+        {
+          gasPrice: gasPrice ? new BigNumber(gasPrice) : undefined,
+          gasLimit: gasLimit
+            ? new BigNumber(gasLimit)
+            : new BigNumber(estimatedGas).multipliedBy(1.5).integerValue()
+        }
+      )
+
+      return {
+        rawTx,
+        path,
+        maxFees,
+        value,
```

Comment:
Im not really sure what this value is here. I think we can not really know what value will be transferred. This is exactly the point of the closeTrustline method, that it will transfer a variable amount to be able to close it. If this is the value from closePath, we should not return it here, because this might not be the actual value that is transferred.